### DISCUSSION TITLE
v10.1.0.3 // Smart Meter Interface // Scripts not working

### DISCUSSION THREAD

**Discussion by RcRaCk2k** [arendst/Tasmota#14192] [Q&A] [answered]
Hi Guys,

i am running Tasmota v10.1.0.3 sensors self compiled with scripts enabled. When installing the default-script provided from an example, i can not see TX-Packets on the Bus-System.

I am using the TTL to RS485 converter board with VCC, TX, RX, GND connections - https://www.reichelt.de/entwicklerboard-uart-ttl-zu-rs485-converter-debo-ttl-rs485-p316262.html

I have a WEMOS D1 mini and connected RX to GPIO4 (D2) and TX to GPIO5 (D1).

**Running script**
`
>D
>B
=>sensor53 r
=>sensor53 d1
>M 1
+1,4,m,0,9600,MODBUS,5,1,01030000,01030004,0103000E,01030010,01030012,01030014,01030016,01030018,0103001A,0103001C,0103001E,01030020,01030022
`

This should send 13 commands to Modbus Device-ID 1 @ 9600b @ 100ms interval on GPIO5.

But the TX-LED on the TTL/MODBUS Converter is not blinking.

But when i configure SDM630TX to GPIO5 i can see the Module-LED blinking.

So i think that the script is not working properly?

The other issue i have seen is, that even if GPIO5 is configured (not NONE) there is no error-message "duplicate GPIO defined" as defined in the documentation! When i use GPIO5 on Tasmota Module Parameters and in the script, that message should appear in the Console, but it does not!

How can i send ModBus Data from Console? I like to change the Device-ID from 0x01 to 0x02 and so on. I have more than just one Smart-Meter on my bus connected.

**Comment by barbudor** [comment]
Try to invert TX and RX
Depending on drivers or boards they may be seen from different perspective...

**Comment by gemu2015** [comment, marked as answer]
did you forgot the definition stop char in your meter definition  ? (#)

\>D
\>B
=>sensor53 r
=>sensor53 d1
\>M 1
+1,4,m,0,9600,MODBUS,5,1,01030000,01030004,0103000E,01030010,01030012,01030014,01030016,01030018,0103001A,0103001C,0103001E,01030020,01030022
\#

this script works in my test device as expected !

**Reply by RcRaCk2k** [reply]
@gemu2015 thank you very much... That was the solution. Yeah i forgot the # on the end of the line.

I was seeing that my Meter definition must have some typos, because the ESP was not sending anything to the BUS. And that was the culprit, but i was not able to find this misspelling by my self. Thank you very much, you are my hero 👍

**Reply by RcRaCk2k** [reply]
@gemu2015 do you also have a solution, how to send commands to the MODBUS via CONSOLE? I need to send the Command 0106000202 to change the Meter-ID. I have 4 Meters on the BUS, so i need to set every Meter to its own ID. What is the best practice to achive that?

**Reply by gemu2015** [reply]
normally MODBUS meters have a local display with a menu that allows to set the MODBUS ID. (at least my SDM530 and SDM630 have that).
otherwise you can connect each single meter after the other and send the set meter id cmd . since this setting should be permanent you then could connect all meters together.

you may send serial cmds via:
1. tasmota serial bridge (see tasmota docs)
2. with scripter SML driver
sml(m 1 htxt) = send SML Hexstring htxt as binary to Meter m (if defined USE_SML_SCRIPT_CMD)
 3. with scripter global serial interface (see scripter docs, if defined USE_SCRIPT_SERIAL)
